### ISSUE TITLE
Active time going up, but down as well?

### ISSUE THREAD

**Issue by Floriferous** [SupremeTechnopriest/react-idle-timer#134]
Creating a new issue after #133 

Hi there, I'm a bit confused about active time calculation.

We'd like to store in our DB the total time a user spends on a certain page, so every 5-10 seconds, we'd like to send the additional time spent to our server, total active time seems like the perfect fit for this.

However, this is the value of getTotalActiveTime() in an interval of 5 seconds:

<img width="392" alt="99378846-33a5b480-28c8-11eb-98e4-31605d0dbc07" src="https://user-images.githubusercontent.com/10428082/100909391-f39c1f80-34cc-11eb-839b-f0ad1dc808cc.png">

I expect these values to be strictly increasing, as I spend more time on this page, on and off. However, as you can see, the value jumps back and forth, most of the time going up, but sometimes going down.

Is this the expected behavior, or is something going wrong?

**Comment by SupremeTechnopriest** [comment]
This is working as expected.  Once your user is idle, the idle timeout is subtracted from the total active time as they were not active during the timeout duration.   If you want the total time on page there is a better way to do it.  You can do:

```js
// Do this once and cache the result
const { startTime } = perfomance.getEntriesByType('navigation')
// Do this each time you want the total time on page
const activeTime = performance.now() - startTime;
```

**Comment by Floriferous** [comment]
I'm not sure I understand what you mean, I want the total `active` time on the page, not the `total` time on the page (including idle).

In the example above I stayed on the same page (i.e. the same mounted `useIdleTimer` hook) and clicked outside of it a few times, going back and forth. What doesn't make sense to me is why the `totalActive` time would _ever_ go down, since I can only spend _more_ time on the page, as time goes on. 

This is the code that I ran to make it really clear, don't you agree that it should always log a value larger or equal than the previous one?

```js
const { getTotalActiveTime } = useIdleTimer();
useEffect(() => {
  setInterval(() => console.log(getTotalActiveTime()), 5000);
}, [])
```

**Comment by SupremeTechnopriest** [comment]
If you are using the default timeout of 20 minutes then yes this does not make sense.  I will look into it.

**Comment by SupremeTechnopriest** [comment]
So I created a sandbox that minimally implements your use case here https://codesandbox.io/s/pedantic-mayer-41utq?file=/src/App.js.

Everything seems to be working as expected.  My guess is that the useIdleTimer hook is being called more than once (maybe by a rerender).  If the hook gets re instantiated, then the starting time resets to the current time and could result in what you are seeing.  

The other thing to note is that if the timeout is hit (by default 20 mins) the value of the timeout is subtracted from the totalActiveTime because the user was not active during the timeout duration.  Depending on when you measure it this would cause the value to go down. 

If you want to paste the entire component that instantiates the useIdleTimer hook I can take a look at it, but all the tests pass and this sandbox proves its not an issue with IdleTimer itself.

**Comment by Floriferous** [comment]
My bad I should've started with a repro! I'm trying to play around with this codesandbox, but I can't get it to start getting idle, it seems like it's active all the time, no matter if I navigate to another tab, or application. Are you seeing this as well? Is this an issue with how this hook works on codesandbox?

I get these logs:

```sh
idle false
5002
idle false
10003
idle false
15003
idle false
20001
```

So yeah I can't really reproduce the issue at the moment.

**Comment by SupremeTechnopriest** [comment]
Idle timer does not currently hook into `pagehide` or `pageshow` to trigger an idle event. Its more designed to let you know when your user has walked away from the computer, not so much for detecting when you background the site.  (Think logging out of a bank account or similar). 

Implementing backgrounding functionality is a bit specific to the use case so I have decided to omit it from the scope of IdleTImer. Theres a lot of considerations to keep in mind such as what do you do if they user is on another tab of the same site.  You would need the timers to be able to communicate with each other.  You could use idle timer as the timing mechanism and implement this functionality on top of it with `postmesage` if thats what you needed.

If you want to see an idle event, you can pass the `timeout` prop and set it to something smaller like `3000`. By default its set to 20 minutes.  Thats why you aren't seeing the idle event in that code sandbox.  Try:

```javascript
const { getTotalActiveTime, isIdle } = useIdleTimer({ timeout: 3000 });
```

Then you should see the page go idle every 3 seconds of non activity.

**Comment by SupremeTechnopriest** [comment]
Maybe I could add a method like `setIdle(bool)` so you can manually set the idle state of the timer.  That would enable you to be able to hook into `pagehide` and `pageshow` and still have the timings be accurate.  Let me know if that is something you would like and I can work on implementing it.
